src/loaders/express/tests/helmet.spec.ts

```diff
@@ -0,0 +1,169 @@
+import helmet from 'helmet'
+import { mocked } from 'ts-jest/utils'
+
+import config from 'src/config/config'
+import featureManager from 'src/config/feature-manager'
+
+import helmetMiddlewares from '../helmet'
+
+describe('helmetMiddlewares', () => {
+  jest.mock('helmet')
+  const mockHelmet = mocked(helmet, true)
+  jest.mock('src/config/feature-manager')
+  const mockFeatureManager = mocked(featureManager, true)
+  jest.mock('src/config/config')
+  const mockConfig = mocked(config, true)
+
+  it('should call the correct helmet functions', () => {
+    mockHelmet.xssFilter = jest.fn().mockImplementation(() => 'xssFilter')
+    mockHelmet.noSniff = jest.fn().mockImplementation(() => 'noSniff')
+    mockHelmet.ieNoOpen = jest.fn().mockImplementation(() => 'ieNoOpen')
+    mockHelmet.dnsPrefetchControl = jest
+      .fn()
+      .mockImplementation(() => 'dnsPrefetchControl')
+    mockHelmet.hidePoweredBy = jest
+      .fn()
+      .mockImplementation(() => 'hidePoweredBy')
+    mockHelmet.referrerPolicy = jest
+      .fn()
+      .mockImplementation(() => 'referrerPolicy')
+    mockHelmet.contentSecurityPolicy = jest
+      .fn()
+      .mockImplementation(() => 'contentSecurityPolicy')
```

Comment:
if you want all these mock implementations for all the tests, they shouldn't be lumped in the first test. since these constitute "setup" for all the tests, they should be in a `beforeAll` function. you should then run `jest.resetAllMocks()` in an `afterEach` block to ensure that the mocks get reset so function call history does not leak across tests.

Reply:
should this be beforeEach instead of beforeAll? otherwise jest.resetAllMocks() in afterEach would clear it after the first test

Reply:
sry I meant `clearAllMocks`